### PR TITLE
Various bugfixes

### PR DESCRIPTION
**Pull request by FFroehlich**

### CODE REVIEW

File: include/amici/edata.h [hunk 1]
```diff
@@ -53,19 +53,19 @@ class ExpData {
      * @param observedEventsStdDev (dimension: nmaxevent x nztrue, row-major)
      */
     ExpData(int nytrue, int nztrue, int nmaxevent,
-            std::vector<realtype> ts,
-            std::vector<realtype> observedData,
-            std::vector<realtype> observedDataStdDev,
-            std::vector<realtype> observedEvents,
-            std::vector<realtype> observedEventsStdDev);
+            std::vector<realtype> const& ts,
```
Comment (dweindl) [inline include/amici/edata.h]: I think in most cases constness interferes here with the move inside the setters (works, but move has no effect).

File: include/amici/exception.h [hunk 2]
```diff
@@ -39,79 +39,85 @@ namespace amici {
...
                     char *demangled = NULL;
                     int status = -1;
                     if (info.dli_sname[0] == '_')
-                        demangled = abi::__cxa_demangle(info.dli_sname, NULL, 0, &status);
-                    snprintf(buf, sizeof(buf), "%-3d %*p %s + %zd\n",
-                             i-2, int(2 + sizeof(void*) * 2), callstack[i],
-                             status == 0 ? demangled :
-                             info.dli_sname == 0 ? symbols[i] : info.dli_sname,
-                             (char *)callstack[i] - (char *)info.dli_saddr);
+                        demangled = abi::__cxa_demangle(info.dli_sname, NULL, 0,
```
Comment (dweindl) [inline include/amici/exception.h]: Maybe we can at some point define a `clang-format` style for coherent formatting. Started in `.clang-format` (#168), but never really used.

File: include/amici/model.h [hunk 3]
```diff
@@ -927,7 +927,14 @@ namespace amici {
...
          */
         int checkFinite(const int N,const realtype *array, const char* fun) const;
-
+        
+        /**
+         * @brief Returns the amici version that was used to compile the model
+         * @return ver amici version string
+         */
+        virtual std::string getAmiciVersion() const {
+            throw AmiException("version not set during compilation");
```
Comment (dweindl) [inline include/amici/model.h]:
```suggestion
            throw AmiException("Version not set during code generation");
```

File: python/amici/sbml_import.py [hunk 4]
```diff
@@ -1107,3 +1109,34 @@ def checkLibSBMLErrors(sbml_doc, show_warnings=False):
...
+        Arguments:
+            sym: symbolic expressions @type sympy.Basic
+
+        Returns:
+
+        Raises:
+            raises SBMLException if an unsupported function is encountered
+    """
+
+    unsupported_functions = [sp.factorial, sp.ceiling]
```
Comment (dweindl) [inline python/amici/sbml_import.py]: Can we add `piecewise` here? Occurred in some models of the benchmark collection.

File: scripts/run-cppcheck.sh [hunk 5]
```diff
@@ -8,7 +8,32 @@ AMICI_PATH=$(cd $SCRIPT_PATH/.. && pwd)
...
+
+
+# suppress header warnings for standard libraries
+grep -v "Cppcheck cannot find all the include files" cppcheck.txt > cppcheck_tmp.txt
+mv cppcheck_tmp.txt cppcheck.txt
+
+grep -v "'AmiVectorArray' does not have a operator=" cppcheck.txt > cppcheck_tmp.txt
+mv cppcheck_tmp.txt cppcheck.txt
+
+grep -v "Member variable 'ExpData::nytrue_' is not initialized in the constructor" cppcheck.txt > cppcheck_tmp.txt
```
Comment (dweindl) [inline scripts/run-cppcheck.sh]: Shouldn't we rather fix those? Either do in every constructor or set to zero in the class definition.

File: src/model.cpp [hunk 6]
```diff
@@ -602,9 +605,11 @@ const std::vector<int> &Model::getParameterList() const {
 }
 
 void Model::setParameterList(const std::vector<int> &plist) {
-    for(auto const& idx: plist)
-        if(idx < 0 || idx >= np())
-            throw AmiException("Indices in plist must be in [0..np]");
+    int np = this->np(); // cannot capture 'this' in lambda expression
+    if(std::any_of(plist.begin(), plist.end(),
```
Comment (dweindl) [inline src/model.cpp]: :+1:

File: include/amici/exception.h (hunk 2 above)
Comment (FFroehlich) [reply include/amici/exception.h]: this was formatted by clang format :D

File: scripts/run-cppcheck.sh (hunk 5 above)
Comment (FFroehlich) [reply scripts/run-cppcheck.sh]: this is a false positive, they are initialized in the delegate constructors which cppcheck does not recognize

File: include/amici/exception.h (hunk 2 above)
Comment (dweindl) [reply include/amici/exception.h]: Ah, I thought the old version was. Alright then ;).

**Comment by dweindl** [comment]
Let's get that merged?

**Pull request merged**